row_cache.cc

```diff
@@ -47,7 +47,8 @@ using namespace cache;
 flat_mutation_reader
 row_cache::create_underlying_reader(read_context& ctx, mutation_source& src, const dht::partition_range& pr) {
     ctx.on_underlying_created();
-    return src.make_reader(_schema, pr, ctx.slice(), ctx.pc(), ctx.trace_state(), streamed_mutation::forwarding::yes);
+    return src.make_reader(_schema, pr, ctx.slice(), ctx.pc(), ctx.trace_state(), streamed_mutation::forwarding::yes,
+            mutation_reader::forwarding(ctx.is_range_query() || ctx.fwd_mr()));
```

Comment:
@psarna this change seems reasonable,  but I wouldn't know if this are the only cases that need mutation-reader fastforward... Did you try to put no here and saw these cases in the tests? Or did you discover them with code review?

Reply:
@psarna ping ^

Reply:
Oh, pardon, missed the question. That one was dictated for me by Tomek, and since we're inside row cache code I thus consider it axiomatic. I initially put the default value here, but Tomek pointed out it's not 100% correct.